### ISSUE TITLE
Method to compare two H5M files

### ISSUE THREAD

**Issue by gonuke** [pyne/pyne#1115]
In some tests (e.g. r2s) we currently compare text files for equivalence.  It would be useful to be able to compare H5M files as well.  This might rely on methods developed in MOAB as well as additions that are PyNE specific.

**Comment by zxkjack123** [comment]
The tool to compare h5 files: `h5diff` is add in #1128 , close this issue.

**Comment by gonuke** [comment]
Do we want a semantic comparison, as well?  This is just a content comparison, but doesn't interpret the differences? Maybe we can discuss this.

**Comment by zxkjack123** [comment]
`h5diff` tells us if two `h5` files are different. If they are different, it tells us 'x different found` in `xxx`. If we use `h5diff -v`, it tells the difference in detail.
This is enough if we want to
- Check whether two `h5` files are exactly the same (entire file content comparison). Using `h5diff f1 f2`.
- check whether two `h5` files have the same objects of contents. Using `h5diff f1 f2 obj1 obj2` For example, just compare tag values, use `h5diff f1 f2 /tstt/tags /tstt/tags`
- Check whether two 'h5` files have almost the same values of some float numbers. Using `h5diff -relative=R f1 f2 obj1 obj2`.

For example, the following is the command I used to compare the `source_density` tag of two `h5m` files:
```
h5diff source_1.h5m source_2.h5m /tstt/elements/Hex8/tags/source_density /tstt/elements/Hex8/tags/source_density
dataset: </tstt/elements/Hex8/tags/source_density> and </tstt/elements/Hex8/tags/source_density>
192507 differences found
```
```
h5diff --relative=0.5 source_1.h5m source_2.h5m /tstt/elements/Hex8/tags/source_density /tstt/elements/Hex8/tags/source_density
dataset: </tstt/elements/Hex8/tags/source_density> and </tstt/elements/Hex8/tags/source_density>
18140 differences found
 ```

**Comment by zxkjack123** [comment]
I did several more tests with `h5diff`, found that sometimes it behaves different as what we want. The `DATASET`, `ATTRIBUTE` `TYPE` make things more complex. Sometimes the return value is different from my expectation.

**Comment by gonuke** [comment]
I agree that this can be closed with the contributions of #1128
